x/evm/statedb/state_object.go

```diff
@@ -0,0 +1,245 @@
+package statedb
+
+import (
+	"bytes"
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var emptyCodeHash = crypto.Keccak256(nil)
+
+// Account is the Ethereum consensus representation of accounts.
+// These objects are stored in the main account trie.
+type Account struct {
+	Nonce    uint64
+	Balance  *big.Int
+	CodeHash []byte
+}
+
+// NewEmptyAccount returns an empty account.
+func NewEmptyAccount() *Account {
+	return &Account{
+		Balance:  new(big.Int),
+		CodeHash: emptyCodeHash,
+	}
+}
+
+// IsContract returns if the account contains contract code.
+func (acct Account) IsContract() bool {
+	return !bytes.Equal(acct.CodeHash, emptyCodeHash)
+}
+
+type Storage map[common.Hash]common.Hash
+
+// SortedKeys sort the keys for deterministic iteration
+func (s Storage) SortedKeys() []common.Hash {
+	keys := make([]common.Hash, len(s))
+	i := 0
+	for k := range s {
+		keys[i] = k
+		i++
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return bytes.Compare(keys[i].Bytes(), keys[j].Bytes()) < 0
+	})
+	return keys
+}
+
+// stateObject is the state of an acount
+type stateObject struct {
+	db *StateDB
+
+	account Account
+	code    []byte
+
+	// state storage
+	originStorage Storage
+	dirtyStorage  Storage
+
+	// DB error.
+	// State objects are used by the consensus core and VM which are
+	// unable to deal with database-level errors. Any error that occurs
+	// during a database read is memoized here and will eventually be returned
+	// by StateDB.Commit.
```

Comment:
no `dbErr`?

Reply:
removed, since our account keeper API doesn't return database errors, so the field is useless here.